neard/tests/node_cluster.rs

```diff
@@ -0,0 +1,104 @@
+use futures::future;
+
+use near_actix_test_utils::{run_actix_until_stop, spawn_interruptible};
+use near_client::{ClientActor, ViewClientActor};
+use near_primitives::types::{BlockHeight, BlockHeightDelta, NumSeats, NumShards};
+use testlib::{start_nodes, test_helpers::heavy_test};
+
+#[derive(Debug, Default)]
+pub struct NodeCluster {
+    dirs: Vec<tempfile::TempDir>,
+    is_heavy: bool,
+    num_shards: Option<NumShards>,
+    num_validator_seats: Option<NumSeats>,
+    num_lightclient: Option<usize>,
+    epoch_length: Option<BlockHeightDelta>,
+    genesis_height: Option<BlockHeight>,
+}
+
+macro_rules! add_mut_helpers {
+    ($($method:ident($self:ident $(,$arg:ident:$type:ty)?) => $expr:expr),+) => {
+        $(
+            pub fn $method(mut $self$(, $arg: $type)?) -> Self {
+                $expr;
+                $self
+            }
+        )+
+    }
+}
+
+impl NodeCluster {
+    pub fn new<F: Fn(usize) -> String>(capacity: usize, gen_dirname: F) -> Self {
```

Comment:
Let’s go with `nodes_count` instead of capacity unless we go with the plan in another comment about lightclint_nodes

Reply:
I think I'll go with this instead of the other option suggested in the comment prior.